File: test/regression/sdk.py

import asyncio
import sys
import os
sys.path.insert(0, os.path.dirname(os.path.abspath(__file__)))

from sdk_trades import open_trade, close_trade, get_trades

if __name__ == "__main__":
    command = sys.argv[1]
    provider = sys.argv[2] if len(sys.argv) > 2 else "https://mainnet.base.org"
    
    if command == "open":
        private_key = sys.argv[3]
        asset_pair = sys.argv[4]
        collateral = sys.argv[5]
        leverage = sys.argv[6]
        asyncio.run(open_trade(private_key, provider, asset_pair, collateral, leverage))
    elif command == "close":
        private_key = sys.argv[3]
        trade_index = int(sys.argv[4])
        pair_index = int(sys.argv[5])
        asyncio.run(close_trade(private_key, provider, trade_index, pair_index))
    elif command == "trades":
        private_key = sys.argv[3]
        asyncio.run(get_trades(private_key, provider))

File: test/regression/sdk.ts
import { exec } from "child_process";
import { promisify } from "util";
import { fileURLToPath } from "url";
import * as path from "path";

const execAsync = promisify(exec);
const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);
const PYTHON_SCRIPT = path.join(__dirname, "sdk.py");

export async function openTrade(
  privateKey: string,
  assetPair: string,
  collateral: string,
  leverage: string
): Promise<any> {
  const { stdout, stderr } = await execAsync(
    `python3 ${PYTHON_SCRIPT} open https://mainnet.base.org ${privateKey} ${assetPair} ${collateral} ${leverage}`
  );
  if (stderr) {
    console.error(`Python stderr: ${stderr}`);
  }
  const lines = stdout.trim().split("\n");
  const jsonLine = lines.find((line) => line.trim().startsWith("{"));
  if (!jsonLine) {
    throw new Error(`No JSON output found. Output: ${stdout}`);
  }
  return JSON.parse(jsonLine.trim());
}

export async function closeTrade(
  privateKey: string,
  tradeIndex: number,
  pairIndex: number
): Promise<any> {
  const { stdout, stderr } = await execAsync(
    `python3 ${PYTHON_SCRIPT} close https://mainnet.base.org ${privateKey} ${tradeIndex} ${pairIndex}`
  );
  if (stderr) {
    console.error(`Python stderr: ${stderr}`);
  }
  const lines = stdout.trim().split("\n");
  const jsonLine = lines.find((line) => line.trim().startsWith("{"));
  if (!jsonLine) {
    throw new Error(`No JSON output found. Output: ${stdout}`);
  }
  return JSON.parse(jsonLine.trim());
}

export async function getTrades(privateKey: string): Promise<any> {
  const { stdout, stderr } = await execAsync(
    `python3 ${PYTHON_SCRIPT} trades https://mainnet.base.org ${privateKey}`
  );
  if (stderr) {
    console.error(`Python stderr: ${stderr}`);
  }
  const lines = stdout.trim().split("\n");
  const jsonLine = lines.find((line) => line.trim().startsWith("{"));
  if (!jsonLine) {
    throw new Error(`No JSON output found. Output: ${stdout}`);
  }
  return JSON.parse(jsonLine.trim());
}
